client/modules/IDE/components/CollectionList/CollectionList.jsx

```diff
@@ -1,313 +1,206 @@
+/* eslint-disable react/react-in-jsx-scope */
 import PropTypes from 'prop-types';
-import React from 'react';
 import { Helmet } from 'react-helmet';
-import { withTranslation } from 'react-i18next';
-import { connect } from 'react-redux';
-import { bindActionCreators } from 'redux';
+import { useDispatch, useSelector } from 'react-redux';
+import { useState, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
 import classNames from 'classnames';
+import { useParams } from 'react-router-dom';
 import { find } from 'lodash';
-import * as ProjectActions from '../../actions/project';
-import * as ProjectsActions from '../../actions/projects';
-import * as CollectionsActions from '../../actions/collections';
-import * as ToastActions from '../../actions/toast';
-import * as SortingActions from '../../actions/sorting';
-import getSortedCollections from '../../selectors/collections';
 import Loader from '../../../App/components/loader';
 import Overlay from '../../../App/components/Overlay';
 import AddToCollectionSketchList from '../AddToCollectionSketchList';
 import { SketchSearchbar } from '../Searchbar';
-
-import CollectionListRow from './CollectionListRow';
-
 import ArrowUpIcon from '../../../../images/sort-arrow-up.svg';
 import ArrowDownIcon from '../../../../images/sort-arrow-down.svg';
+import { getProject } from '../../actions/project';
+import { getCollections } from '../../actions/collections';
+import getSortedCollections from '../../selectors/collections';
+import { DIRECTION, toggleDirectionForField } from '../../actions/sorting';
+import CollectionListRow from './CollectionListRow';
 
-class CollectionList extends React.Component {
-  constructor(props) {
-    super(props);
-
-    if (props.projectId) {
-      props.getProject(props.projectId);
-    }
+const CollectionList = (props) => {
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+  const { project_id: projectId } = useParams() || { project_id: null };
+  const [hasLoadedData, setHasLoadedData] = useState(false);
 
-    this.props.getCollections(this.props.username);
-    this.props.resetSorting();
+  const { user, sorting, loading, collections } = useSelector((state) => ({
+    user: state.user,
+    sorting: state.sorting,
+    loading: state.loading,
+    collections: getSortedCollections(state)
+  }));
```

Comment:
It is better to use individual selectors for each variable because it reduces unnecessary re-renders.  More info at https://react-redux.js.org/api/hooks#equality-comparisons-and-updates